functional_tests/functional/functional_test.go

```diff
@@ -344,11 +343,11 @@ func deployChartsAndApps(t *testing.T, testKubeConfig string) {
 			return
 		}
 		t.Log("Cleaning up cluster")
-		teardown(t, testKubeConfig)
+		teardown(t, context.Background(), testKubeConfig)
```

Comment:
This will end up needing a `//nolint` directive. The current output of linting is:
```
context.Background() could be replaced by t.Context() in ... (usetesting)
```
I originally tried to replace `context.Background()` in this instance, but since this is called in test cleanup, the test's context is already cancelled. This means that following usages of `t.Context()` always return context cancelled, so cleanup actions fail, causing functional tests to fail.